app/models/accounting/transaction.rb

```diff
@@ -95,4 +112,27 @@ def lookup_currency_id
       project.currency_id
     end
   end
+
+  def calculate_loan_amounts
+    loan_id = self.project_id
+
+    if self.loan_transaction_type == 'disbursement'
+      self.update_column(:change_in_principal, amount)
```

Comment:
I think you should just be assigning these values and then updating the column once at the end (preferably by just calling `save!` rather than `update_column`, right now this would do the update as a bunch of different database queries.